test/user.test.js

```diff
@@ -2370,6 +2370,28 @@ describe('User', function() {
       });
     });
 
+    // See https://github.com/strongloop/loopback/issues/3215
+    it('handles subclassed user with no accessToken relation', () => {
+      // setup a new LoopBack app, we don't want to use shared models
+      app = loopback({localRegistry: true, loadBuiltinModels: true});
+      app.set('remoting', {errorHandler: {debug: true, log: false}});
+      app.dataSource('db', {connector: 'memory'});
+      const User = app.registry.createModel({
+        name: 'user',
+        base: 'User',
+      });
+      app.model(User, {dataSource: 'db'});
+      app.enableAuth({dataSource: 'db'});
+      expect(app.models.User.modelName).to.eql('user');
+
+      return User.create(validCredentials)
+        .then(u => {
+          u.email = 'updated@example.com';
+          return u.save();
+          // the test passes when save() does not throw any error
```

Comment:
>  What I'm suggesting is we should have something that catches and calls expect.fail() instead of the comment.

By calling `expect.fail`, the test would hide the original error and report an unhelpful failure.

Note that we are asserting a resolved promise here, not a rejected promise as is described in the style guide link you pasted.

I suppose I could rewrite this test as follows, would you find it easier to read?

```js
.then(u => {
  u.email = 'updated@example.com';
  return u.save();
})
.then(u => {
  // the test passes when save() does not throw any error
});
```